File: php-pr/51/php/code-quality-error-handling/SecureDatabaseConnector.php
<?php

class SecureDatabaseConnector
{
    private $host = 'localhost';
    private $username = 'root';
    private $password = 'password';
    private $database = 'testdb';

    public function connect()
    {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        try {
            $pdo = new PDO("mysql:host={$this->host};dbname={$this->database}", 
                          $this->username, $this->password);
            $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
            return $pdo;
        } catch (PDOException $e) {
            error_log("Database connection failed: " . $e->getMessage());
            throw new RuntimeException("Unable to connect to database", 0, $e);
        }
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-error-handling/secure_database_connector.rb
require 'sqlite3'

class SecureDatabaseConnector
  def initialize
    @host = 'localhost'
    @database = 'testdb'
  end

  def connect
    # {fact rule=code-quality-error-handling@v1.0 defects=0}
    begin
      db = SQLite3::Database.new(@database)
      return db
    rescue SQLite3::Exception => e
      puts "Database connection failed: #{e.message}"
      raise RuntimeError, "Unable to connect to database: #{e.message}"
    rescue StandardError => e
      puts "Unexpected error: #{e.message}"
      raise
    end
    # {/fact}
  end
end